package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless

```diff
@@ -53,7 +53,7 @@ local function is_outdoor()
 end
 
 local function get_channel(radio, config)
-	if radio.band == '5g' and is_outdoor() then
+	if radio.band == '5g' and (is_outdoor() or not util.supports_channel(radio, config.channel())) then
```

Comment:
I'm wondering if something like the following would make sense:
```lua
	if (radio.band == '5g' and is_outdoor()) or not util.supports_channel(radio, config.channel()) then
```
But I guess having radios for distinct ranges isn't really a thing for 2.4GHz...

Reply:
Yes, I think that your suggestion still makes sense - I adapted it.